doc/source/serve/model-multiplexing.md

```diff
@@ -0,0 +1,70 @@
+# Model Multiplexing
+
+This section helps you understand how to write multiplexed deployment by using the `serve.multiplexed` and `serve.get_multiplexed_model_id` APIs.
+
+This is an experimental feature and the API may change in the future. You are welcome to try it out and give us feedback!
+
+## Why model multiplexing?
+
+Model multiplexing is a technique for serving multiple models in a single replica. Traffic is routed to the corresponding model based on the request header. To serve multiple models with a pool of replicas to optimize cost. This is useful in cases where you might have many models with the same shape but different weights that are sparsely invoked. If any replica for the deployment has the model loaded, incoming traffic for that model (based on request header) will automatically be routed to that replica avoiding unnecessary load time.
+
+## Wrting a multiplexed deployment
+
+To write a multiplexed deployment, use the `serve.multiplexed` and `serve.get_multiplexed_model_id` APIs.
+
+Assuming you have multiple Torch models inside aws s3 bucket with the following structure:
+```
+s3://my_bucket/1/model.pt
+s3://my_bucket/2/model.pt
+s3://my_bucket/3/model.pt
+s3://my_bucket/4/model.pt
+...
+```
+
+Define a multiplexed deployment:
+```{literalinclude} doc_code/multiplexed.py
+:language: python
+:start-after: __serve_deployment_example_begin__
+:end-before: __serve_deployment_example_end__
+```
+
+:::{note}
+The `serve.multiplexed` API also has a `max_num_models_per_replica` parameter. Use it to configure how many models to load in a single replica. If the number of models is larger than `max_num_models_per_replica`, Serve uses the LRU policy to evict the least recently used model.
+:::
+
+:::{tip}
+This code example uses the Pytorch Model object. You can also define your own model class and use it here. Make sure the model class has a `__call__` method. To release resources when the model is evicted, implement the `__del__` method. Ray Serve internally calls the `__call__` method to execute the model, and calls the `__del__` method to release resources when the model is evicted.
```

Comment:
What is the __call__ method requirement in the model class?

Reply:
good catch